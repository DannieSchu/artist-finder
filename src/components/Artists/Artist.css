.Artist {
  width: 100%;
  padding: 1rem;
  
  th {
    color: rgb(160, 231, 243);
    background-color: rgb(22, 53, 88);
    border-radius: 50% 20% / 10% 40%;
    text-align: center;
    width: 3rem;
    height: 1.5rem;
    padding: 1rem 1rem;
  }

  td {
    color: rgb(22, 53, 88);
    text-align: center;
    width: 3rem;
    height: 1.5rem;
    padding: 1rem 1rem;
  } 
}