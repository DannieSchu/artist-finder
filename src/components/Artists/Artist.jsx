import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../global-styles/item.css';

const Artist = ({ name, origin, birth, death }) => (
  <section className={styles.item}>
    <tr>
      <th>Artist</th>
      <td>{name}</td>
    </tr>
    <tr>
      {origin && <td>{origin}</td>}
      {origin && <th>Origin</th>}
    </tr>
    <tr>
      {birth && <th>DOB</th>}
      {birth && <td>{birth.slice(0, 4)}</td>}
    </tr>
    <tr>
      {death && <td>{death.slice(0, 4)}</td>}
      {death && <th>Death</th>}
    </tr>
  </section>
);

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  origin: PropTypes.string,
  birth: PropTypes.string,
  death: PropTypes.string
};

export default Artist;
