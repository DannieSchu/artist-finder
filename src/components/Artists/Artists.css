.Artists {
  display: flex;
  flex-direction: column;
  align-items: center;

  ul {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
    width: 100%;
  }

  li {
    color: rgb(160, 231, 243);
    display: flex;
    flex-direction: column;
    list-style-type: none;
    margin: 1rem;
    padding: 1rem;
    border-radius: 10% 40% / 50% 20%;
    min-width: 13rem;
    min-height: 10rem;
    background-color: rgb(22, 53, 88);
    background-color: rgb(127, 226, 251);
  }
}