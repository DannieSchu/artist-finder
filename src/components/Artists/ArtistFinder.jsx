import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist.jsx';
import Header from '../Header/Header.jsx';
import Home from '../Home/Home.jsx';
import { useFindArtist } from '../../hooks/findArtist.jsx';
import withPaging from '../../utils/withPaging.jsx';
import withList from '../../utils/withList.jsx';

const ArtistFinder = ({ page, setTotalPages }) => {
  const { artists, loading, artistSearch, handleChange, handleSubmit } = useFindArtist({ page, setTotalPages });
  const Artists = withList(Artist, 'artists');

  return (
    <>
      <Header 
        onChange={handleChange} 
        onSubmit={handleSubmit} 
        value={artistSearch} 
        placeholder="Find artist" 
        buttonText="Go!"
      />
      {artists.length === 0 && <Home />}
      {artists.length !== 0 && <Artists artists={artists} heading={artistSearch} loading={loading} />}
    </>
  );
};

ArtistFinder.propTypes = {
  page: PropTypes.number.isRequired,
  setTotalPages: PropTypes.func.isRequired
};

export default withPaging(ArtistFinder);
