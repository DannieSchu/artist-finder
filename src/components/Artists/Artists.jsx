import PropTypes from 'prop-types';
import Artist from './Artist.jsx';
import withList from '../../utils/withList.jsx';

const Artists = withList(Artist, 'artists');

Artists.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    origin: PropTypes.string,
    birth: PropTypes.string,
    death: PropTypes.string,
  })).isRequired,
  loading: PropTypes.bool
};

export default Artists;
