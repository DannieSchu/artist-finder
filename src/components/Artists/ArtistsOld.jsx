import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Artist from './Artist.jsx';
import styles from './Artists.css';

const Artists = ({ artists, loading }) => {
  const artistElements = artists.map(artist => (
    <li key={artist.id}>
      <Link to={`/releases/${artist.id}`}>
        <Artist {...artist} />
      </Link>
    </li>
  ));

  return (
    <section className={styles.Artists}>
      {loading && <h2>Loading...</h2>}
      <h2>Artists</h2>
      <ul>
        {artistElements}
      </ul>
    </section>
  );
};

Artists.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    origin: PropTypes.string,
    birth: PropTypes.string,
    death: PropTypes.string,
  })).isRequired,
  loading: PropTypes.bool
};

export default Artists;
