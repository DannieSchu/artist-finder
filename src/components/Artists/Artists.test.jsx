import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists.jsx';

describe('Artists', () => {
  it('matches a snapshot', () => {
    const artists = [{
      id: 'gh8932hg3gee',
      name: 'Leonard Cohen',
      origin: 'Canada',
      birth: '1934-09-21' },
    {
      id: '6943w0utw0ghjvdr',
      name: 'Leonard Cohen',
      origin: 'Canada',
      birth: '1934-09-21' },
    {
      id: 't430tu340grjid',
      name: 'Leonard Cohen',
      origin: 'Canada',
      birth: '1934-09-21' },
    ];
    
    const wrapper = shallow(
      <Artists artists={artists} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
