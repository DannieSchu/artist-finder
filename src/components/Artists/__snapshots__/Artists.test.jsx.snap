// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists matches a snapshot 1`] = `
<section
  className="Artists"
>
  <h1>
    Artists
  </h1>
  <ul>
    <li
      key="gh8932hg3gee"
    >
      <Link
        to="/gh8932hg3gee"
      >
        <Artist
          birth="1934-09-21"
          death="1934-09-21"
          id="gh8932hg3gee"
          name="Leonard Cohen"
          origin="Canada"
        />
      </Link>
    </li>
    <li
      key="6943w0utw0ghjvdr"
    >
      <Link
        to="/6943w0utw0ghjvdr"
      >
        <Artist
          birth="1934-09-21"
          death="1934-09-21"
          id="6943w0utw0ghjvdr"
          name="Leonard Cohen"
          origin="Canada"
        />
      </Link>
    </li>
    <li
      key="t430tu340grjid"
    >
      <Link
        to="/t430tu340grjid"
      >
        <Artist
          birth="1934-09-21"
          death="1934-09-21"
          id="t430tu340grjid"
          name="Leonard Cohen"
          origin="Canada"
        />
      </Link>
    </li>
  </ul>
</section>
`;
